src/axom/sidre/core/View.cpp

```diff
@@ -630,17 +647,43 @@ int View::getShape(int ndims, IndexType* shape) const
   }
 
   // Fill the rest of the array with zeros (when ndims > shapeSize)
-  if(ndims > shapeSize)
+  for(int i = shapeSize ; i < ndims ; ++i)
   {
-    for(int i = shapeSize ; i < ndims ; ++i)
-    {
-      shape[i] = 0;
-    }
+    shape[i] = 0;
   }
 
   return m_shape.size();
 }
 
+/*
+ *************************************************************************
+ *
+ * Return number of dimensions and fill in permutation information.
+ *
+ *************************************************************************
+ */
+int View::getPermutation(int ndims, IndexType* permutation) const
+{
+  if (static_cast<unsigned>(ndims) < m_permutation.size())
+  {
+    return -1;
+  }
+
+  const int permutationSize = getNumDimensions();
+  for(int i = 0 ; i < permutationSize ; ++i)
+  {
+    permutation[i] = m_permutation[i];
+  }
+
+  // Fill the rest of the array with -1 (when ndims > permutationSize)
+  for(int i = permutationSize ; i < ndims ; ++i)
+  {
+    permutation[i] = -1;
```

Comment:
This seems a bit weird as well -- should it be an error if ``ndims != getNumDimensions()``?

Otherwise, users need to know about the semantics of a permutation entry that has value ``-1``.

Reply:
Again this is just mimicking the semantics of `View::getShape`. There the extra values are filled with zeros which is not a valid dimension size. Here zero is a valid permutation entry so I chose to fill them with -1.

I'm in favor of dropping the `ndims` argument altogether and just assume that the pointer has enough space. We could also add a `C++` only method `std::vector<IndexType> View::getShape() const`. However I feel that changing up the semantics of `getShape` is out of scope for this PR.